src/ServiceControl.Plugin.Nsb6.SagaAudit/CaptureSagaResultingMessagesBehavior.cs

```diff
@@ -1,68 +1,69 @@
-                {
-                    
-                    ResultingMessageId = context.OutgoingMessage.Id,
-                    TimeSent = DateTimeExtensions.ToUtcDateTime(context.OutgoingMessage.Headers[Headers.TimeSent]),
-                    MessageType = logicalMessage.MessageType.ToString(),
-                    DeliveryDelay = sendOptions.DelayDeliveryWith,
-                    DeliveryAt = sendOptions.DeliverAt,
-                    Destination = sendOptions.Destination.ToString(),
-                    Intent = "Send" //TODO: How to get the proper message intent!?
-                };
-                sagaUpdatedMessage.ResultingMessages.Add(sagaResultingMessage);
+                deliveryDelay = delayDeliveryWith.Delay;
             }
 
-            if (context.DeliveryOptions is PublishOptions)
+            DateTime? doNotDeliverBefore = null;
+            DoNotDeliverBefore notDeliverBefore;
+            if (context.Extensions.TryGetDeliveryConstraint(out notDeliverBefore))
             {
-                var sagaResultingMessage = new SagaChangeOutput
-                {
-                    ResultingMessageId = context.OutgoingMessage.Id,
-                    TimeSent = DateTimeExtensions.ToUtcDateTime(context.OutgoingMessage.Headers[Headers.TimeSent]),
-                    MessageType = logicalMessage.MessageType.ToString(),
-                    //TODO: Can we remove the DeliveryDelay and DeliveryAt here??
-                    //DeliveryDelay = publishOptions.DelayDeliveryWith,
-                    //DeliveryAt = publishOptions.DeliverAt,
-                    //Destination = GetDestination(context),
-                    Intent = "Publish" //TODO: get the message intent the right way!
-                };
-                sagaUpdatedMessage.ResultingMessages.Add(sagaResultingMessage);
+                doNotDeliverBefore = notDeliverBefore.At;
             }
+
+            var sagaResultingMessage = new SagaChangeOutput
+            {
+                ResultingMessageId = context.MessageId,
+                TimeSent = DateTimeExtensions.ToUtcDateTime(context.Headers[Headers.TimeSent]),
+                MessageType = logicalMessage.MessageType.ToString(),
+                DeliveryDelay = deliveryDelay,
+                DeliveryAt = doNotDeliverBefore,
+                Destination = GetDestinationForUnicastMessages(context),
+                Intent = context.Headers[Headers.MessageIntent]
+            };
+            sagaUpdatedMessage.ResultingMessages.Add(sagaResultingMessage);
+        }
+
+        SagaUpdatedMessage sagaUpdatedMessage;
+
+        static string GetDestinationForUnicastMessages(IOutgoingLogicalMessageContext context)
+        {
+            var sendAddressTags = context.RoutingStrategies.OfType<UnicastRoutingStrategy>().Select(urs => urs.Apply(context.Headers)).Cast<UnicastAddressTag>().ToList();
+            return sendAddressTags.Count() != 1 ? null : sendAddressTags.First().Destination;
```

Comment:
When the `OutgoingLogicalMessageContext` is created the routing strategies are populated. Send uses UnicastRoutingStrategy. It is a little ugly but by casting it we can dig the destination out.

I am not sure how this "smells" from a core perspective and I am sure this enumeration is brittle but I wanted to get the thought in there.